controller/linodemachine_controller.go

```diff
@@ -310,41 +317,48 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return linodeInstance, nil
 }
 
-func (r *LinodeMachineReconciler) configureDisksControlPlane(
+func (r *LinodeMachineReconciler) configureInstanceProfile(
 	ctx context.Context,
 	logger logr.Logger,
 	machineScope *scope.MachineScope,
 	linodeInstanceID int,
+	createOpts linodego.InstanceCreateOptions,
 ) error {
 	if !kutil.IsControlPlaneMachine(machineScope.Machine) {
 		return nil
 	}
-	// get the default instance config
-	configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, linodeInstanceID, &linodego.ListOptions{})
-	if err != nil || len(configs) == 0 {
-		logger.Error(err, "Failed to list instance configs")
-
-		return err
-	}
-	instanceConfig := &configs[0]
 
-	// carve out space for the etcd disk
-	rootDiskID := instanceConfig.Devices.SDA.DiskID
-	rootDisk, err := machineScope.LinodeClient.GetInstanceDisk(ctx, linodeInstanceID, rootDiskID)
+	instanceType, err := machineScope.LinodeClient.GetType(ctx, createOpts.Type)
 	if err != nil {
-		logger.Error(err, "Failed to get root disk for instance")
+		logger.Error(err, "Failed to retrieve type for instance")
 
 		return err
 	}
-	diskSize := rootDisk.Size - defaultEtcdDiskSize
-	if err = machineScope.LinodeClient.ResizeInstanceDisk(ctx, linodeInstanceID, rootDiskID, diskSize); err != nil {
-		logger.Error(err, "Failed to resize root disk")
+
+	// create the root disk
+	rootDisk, err := machineScope.LinodeClient.CreateInstanceDisk(
+		ctx,
+		linodeInstanceID,
+		linodego.InstanceDiskCreateOptions{
+			Label:           "root",
+			Size:            instanceType.Disk - defaultEtcdDiskSize,
+			Image:           createOpts.Image,
+			RootPass:        createOpts.RootPass,
+			Filesystem:      string(linodego.FilesystemExt4),
+			AuthorizedKeys:  createOpts.AuthorizedKeys,
+			AuthorizedUsers: createOpts.AuthorizedUsers,
+			StackscriptID:   createOpts.StackScriptID,
+			StackscriptData: createOpts.StackScriptData,
+		},
+	)
+	if err != nil {
+		logger.Error(err, "Failed to create root disk")
 
 		return err
 	}
 
```

Comment:
[Similar to the resize wait that I recently added](https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodemachine_controller.go#L349), we might want to wait for the root and etcd disks to be ready?

Reply:
Ah, sorry I missed that change. I'll add it as well